src/scenes/ModulesRouter/ModuleEvent.tsx

```diff
@@ -0,0 +1,64 @@
+import React, { useMemo } from 'react';
+import { useParams } from 'react-router-dom';
+
+import { Card, Grid, Typography, useTheme } from '@mui/material';
+
+import { Crumb } from 'src/components/BreadCrumbs';
+import PageWithModuleTimeline from 'src/components/common/PageWithModuleTimeline';
+import { APP_PATHS } from 'src/constants';
+import { useLocalization } from 'src/providers';
+import { selectModule } from 'src/redux/features/modules/modulesSelectors';
+import { useAppSelector } from 'src/redux/store';
+
+import ModuleViewTitle from './ModuleViewTitle';
+
+const ModuleEvent = () => {
+  const { activeLocale } = useLocalization();
+  const theme = useTheme();
+  const pathParams = useParams<{ eventId: string; moduleId: string; projectId: string }>();
+  const eventId = Number(pathParams.eventId);
+  const moduleId = Number(pathParams.moduleId);
+  const projectId = Number(pathParams.projectId);
+  const module = useAppSelector(selectModule(moduleId));
+  const event = module?.events.find((e) => e.id === eventId);
```

Comment:
```
  const event = module?.events.find((e) => e.id === eventId);
```

Perhaps this should be memoized?

Reply:
Might be a good idea to memoize, theoretically the array can be large in size to search.